screens/user-list.tsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * Sample React Native App
+ * https://github.com/facebook/react-native
+ *
+ * Generated with the TypeScript template
+ * https://github.com/react-native-community/react-native-template-typescript
+ *
+ * @format
+ */
+import React, { useState } from 'react';
+import { View, Text, FlatList, StyleSheet, ActivityIndicator } from 'react-native';
+import { NetworkStatus, useQuery } from '@apollo/client';
+import { getUsers } from '../components/utils/graphql-requests';
+
+const UserListScreen = () => {
+  const [offset, setOffset] = useState(0);
+
+  const { loading, error, data, refetch, networkStatus } = useQuery(getUsers, {
+    variables: { offset: offset },
+    notifyOnNetworkStatusChange: true,
+  });
+
+  const [users, setUser] = useState(data?.users.nodes);
+
+  const UpdateOffset = () => {
+    const last = data.users.nodes.length - 1;
+
+    if (data?.users.nodes[last].id < data?.users.count) {
+      setOffset(offset + 10);
+      refetch({
+        offset: offset,
+      });
```

Comment:
2 coisas sobre esses 2 comandos:

1. No `useQuery`, quando você troca o valor de um parâmetro que está sendo enviado nas variables, ele automaticamente já refaz a query para aquele novo valor. Então vc não precisa trocar o estado e fazer refetch ao mesmo tempo.

2. A sua intenção aqui era incrementar o offset em 10 e fazer o refect com o novo valor do offset? Por que se era isso, provavelmente não vai dar certo. Por questões de otimização, o react faz o que eles chamam "batch update" nos estados. Ou seja, no caso desse callback, ele espera rodar toda a função, ai ele vê quais foram todas as alterações de estado que ocorreram, e aplica tudo de uma vez. acho que vale a pena vc dar uma pesquisadinha sobre isso. Mas o importante é que, se o offset for 10 antes da linha 29, ele ainda será 10 da linha 30 em diante.